generator/final_processor_intermediate_mwm.cpp

```diff
@@ -0,0 +1,516 @@
+                                                     AffiliationInterface const & affilation,
+                                                     size_t threadsCount)
+{
+  computational::ThreadPool pool(threadsCount);
+  std::vector<std::future<std::vector<std::string>>> futuresAffilations;
+  for (auto const & fb : fbs)
+  {
+    auto result = pool.Submit([&]() {
+      return affilation.GetAffiliations(fb);
+    });
+    futuresAffilations.emplace_back(std::move(result));
+  }
+
+  std::vector<std::vector<std::string>> resultAffilations;
+  resultAffilations.reserve(futuresAffilations.size());
+  for (auto & f : futuresAffilations)
+    resultAffilations.emplace_back(f.get());
+
+  return resultAffilations;
+}
+
+template <class SerializationPolicy = MaxAccuracy>
+std::vector<std::vector<std::string>> AppendToCountries(std::vector<FeatureBuilder> const & fbs,
+                                                        std::string const & temproryMwmPath,
+                                                        AffiliationInterface const & affilation,
+                                                        size_t threadsCount)
+{
+  auto const affilations = GetAffilations(fbs, affilation, threadsCount);
+  std::unordered_map<std::string, std::vector<FeatureBuilder>> countryToCities;
+  for (size_t i = 0; i < fbs.size(); ++i)
+  {
+    for (auto const & country : affilations[i])
+      countryToCities[country].emplace_back(fbs[i]);
+  }
+
+  delayed::ThreadPool pool(threadsCount, delayed::ThreadPool::Exit::ExecPending);
+  for (auto & p : countryToCities)
+  {
+    pool.Push([temproryMwmPath, name{p.first}, countries{std::move(p.second)}]() {
+      auto const path = base::JoinPath(temproryMwmPath, name + DATA_FILE_EXTENSION_TMP);
+      FeatureBuilderWriter<SerializationPolicy> collector(path, FileWriter::Op::OP_APPEND);
+      for (auto && fb : countries)
+        collector.Write(std::move(fb));
+    });
+  }
+
+  return affilations;
+}
+
+void Sort(std::vector<FeatureBuilder> & fbs)
+{
+  std::sort(std::begin(fbs), std::end(fbs), [](auto const & l, auto const & r) {
+    auto const lGeomType = static_cast<int8_t>(l.GetGeomType());
+    auto const rGeomType = static_cast<int8_t>(r.GetGeomType());
+
+    auto const lId = l.HasOsmIds() ? l.GetMostGenericOsmId() : base::GeoObjectId();
+    auto const rId = r.HasOsmIds() ? r.GetMostGenericOsmId() : base::GeoObjectId();
+
+    auto const lPointCount = l.GetPointsCount();
+    auto const rPointCount = r.GetPointsCount();
+
+    auto const lKeyPoint = l.GetKeyPoint();
+    auto const rKeyPoint = r.GetKeyPoint();
+
+    return std::tie(lGeomType, lId, lPointCount, lKeyPoint) < std::tie(rGeomType, rId, rPointCount, rKeyPoint);
+  });
+}
+
+bool FilenameIsCountry(std::string filename, AffiliationInterface const & affilation)
+{
+  strings::ReplaceLast(filename, DATA_FILE_EXTENSION_TMP, "");
+  return affilation.HasRegionByName(filename);
+}
+
+class CityBoundariesHelper
```

Comment:
s/City/Cities/
the choice of plurals in CitiesBoundaries (vs CityBoundary) was deliberate

Reply:
++